step-capstone/src/scripts/Suggestion.js

```diff
@@ -0,0 +1,77 @@
+import queryPlaces from './PlacesQuery';
+import categories from './Categories';
+
+const query = (coordinates, service) => {
+    // return results: a map with key : place_id, value: PlaceObject
+    let places = queryPlaces(coordinates, radius, service, ["tourist_attraction", "natural_feature"])
+    places.then(results => {
+        return results;
+    })
+}
+
+const countCat = (placeObject, userCat) => {
+    // Count the number of categories a place fits into
+    let cats = 0;
+    userCat.forEach(catString => {
+        placeObject.place.types.forEach(type => {
+            if (categories.get(catString).has(type)) {
+                cats += 1;
+                break;
+            }
+        })
+    
+    })
+    return cats;
+}
+
+const getScore = (placeCat, userCat, prominence, placePrice, userBudget, rating) => {
+    // return the score for a PlaceObject
+    var catScore = 0;
+    if (cat !== 0) {
+        catScore = 60 + 40/userCat * placeCat;
+    }
+
+    var prominenceScore = (prominence.total - prominence.index)**2/(prominence.total)**2 * 100;
+    
+    var budgetScore = 0;
+    if (placePrice === null) {
+        budgetScore = 50;
+    } else if (userBudget < placePrice) {
+        budgetScore = 0;
+    } else {
+        budgetScore = 100;
+    }
+
+    var ratingScore = rating* 20;
```

Comment:
Same here, comments to explain what this calculates would be useful

Reply:
fixed